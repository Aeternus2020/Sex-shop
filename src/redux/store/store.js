import { applyMiddleware, combineReducers, createStore } from "redux";
import products from "../getProducts/reducers";
import basket from "../pages/basket/reducers";
import favorite from "../pages/favorite/reducers";
import showModal from "../components/modal/reducers"; 
import title from "../pages/app/reducers";
import { composeEnhancers, middleware } from "./middleware";

export const rootReducer = combineReducers({
    products,
    basket,
    favorite,
    showModal,
    title
})

export const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(...middleware))
    )