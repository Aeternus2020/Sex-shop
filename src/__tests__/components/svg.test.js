import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import { store } from "../../redux/store/store";
import { Provider } from "react-redux";
import StarSolid from "../../pages/favorite/star";
import renderer from 'react-test-renderer';

afterEach(cleanup);

test('Test  Favorite star', () => {
    const item = {
        name: "Whip", price: "180$", url: "./images/whip.webp", article: "52169875", color: "black"
    }
    const favoriteArray = [
        {name: "Whip", price: "180$", url: "./images/whip.webp", article: "52169875", color: "black"},
        {name: "Pasties", price: "100$", url: "./images/pasties.webp", article: "75213699", color: "gold"},
        {name: "Bondage", price: "70$", url: "./images/bondage.webp", article: "25987456", color: "black"}
    ]
    render(
        <Provider store={store} >
            <StarSolid id={8} star='#000' item={item} favoriteArray={favoriteArray}/>
        </Provider>
        );
    expect(screen.getByTestId('favoriteStar-8')).toBeInTheDocument();
    expect(screen.getByTestId('favoriteStar-8')).toHaveAttribute('width', '25');
    expect(screen.getByTestId('favoriteStar-8')).toHaveAttribute('height', '25');
    expect(screen.getByTestId('favoriteStar-8')).toHaveClass('svg');
    fireEvent.click(screen.getByTestId('favoriteStar-8'));
})

test('Test snapshot Svg', () => {
    const tree = renderer.create(
    <Provider store={store} >
        <StarSolid id={8}/>
    </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})