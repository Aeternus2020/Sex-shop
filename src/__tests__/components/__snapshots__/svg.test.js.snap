// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Svg 1`] = `
<svg
  className="svg"
  data-testid="favoriteStar-8"
  height="25"
  onClick={[Function]}
  viewBox="0 0 20 20"
  width="25"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M8.565 4.29c.531-1.276 2.34-1.276 2.871 0l.906 2.18a.64.64 0 0 0 .54.392l2.353.189c1.378.11 1.937 1.83.887 2.73l-1.793 1.535a.64.64 0 0 0-.206.635l.548 2.296c.32 1.345-1.143 2.408-2.323 1.687l-2.014-1.23a.64.64 0 0 0-.667 0l-2.014 1.23c-1.18.721-2.643-.342-2.323-1.687l.548-2.296a.64.64 0 0 0-.206-.635L3.879 9.781c-1.05-.9-.49-2.62.888-2.73l2.352-.189a.64.64 0 0 0 .54-.392l.906-2.18Z"
    fillRule="evenodd"
  />
</svg>
`;
